docs/user-guide/deployments.md

```diff
@@ -454,6 +454,142 @@ nginx-deployment-3066724191   0         0         1h
+
+* limit ranges
+
+* application runtime misconfiguration
+
+In those cases you can specify a timeout parameter in the spec ([spec.progressDeadlineSeconds](#progress-deadline-seconds))
+that denotes the seconds you want to wait for your Deployment to report any progress. Once the deadline is exceeded, meaning
+your Deployment will not report any progress for more than progressDeadlineSeconds, the deployment controller will add a
+condition with `Type=Progressing`, `Status=False`, and `Reason=ProgressDeadlineExceeded` in the status of the Deployment.
+For more information on status conditions, see [here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/devel/api-conventions.md#typical-status-properties). 
+Note that at the moment, Kubernetes will do nothing more than just report with a condition with `Reason=ProgressDeadlineExceeded`.
+
+Another thing to note is transient errors eg. your rollout times out and a condition with `Reason=ProgressDeadlineExceeded`
+is set to your Deployment because of insufficient quota. To fix this issue, scale down your Deployment, or scale down other
+controllers you may be running, or increase quota in your namespace. You can increase quota by either asking your administrator
+or doing it yourself assuming you have the right permissions. Once the deployment controller notices that your Deployment is
+complete, it will transition the status of the Progressing condition into a succeeded one (`Status=True` and
+`Reason=NewReplicaSetAvailable`).
+
+A complete Deployment for the controller is one that: 
+
+* has minimum availability, meaning its available replicas are more than those required by the Deployment strategy 
+(`status.availableReplicas >= spec.replicas - maxUnavailable` where `maxUnavailable` is the maximum unavailable replicas
+that a Deployment allows at any point in time. It depends on the deployment strategy: Recreate deployments don't use it whereas
+Rolling Update deployments specify it in [spec.strategy.rollingUpdate.maxUnavailable](#max-unavailable)).
+
+* has all of its replicas updated to the latest revision (`spec.replicas == status.updatedReplicas`).
+
+Note that clients should always take into account generation/observedGeneration for ensuring that the latest state of a
+Deployment has been observed by the controller. For example, `kubectl rollout status` expects the Deployment generation
+to equal the observed generation (`d.generation <= d.status.observedGeneration`) before reporting any information provided
+in the status of the Deployment.
+
+Progress for a Deployment is:
+
+* the creation of the new replica set.
+
+* scaling up new pods.
+
+* scaling down old pods.
+
+To make the controller report lack of progress for a Deployment after 10 minutes:
+
+```shell
+$ kubectl patch deployment/nginx-deployment -p '{"spec":{"progressDeadlineSeconds":600}}'
+"nginx-deployment" patched
+```
+
+If you describe the Deployment, you will notice a Conditions section:
```

Comment:
This section is about how to detect and deal with a stalled Deployment. It isn't necessary to explain what a successful deployment looks like first. Remove everything from between lines 516 and 530.